WIP: Transpose

earboxer opened the pull request:
Based on code and discussion from https://github.com/rism-ch/verovio/issues/1189.

lpugin commented:
Could you make the PR to develop-transpose for now? Also, there is an error with AppVeyor. Any thoughts?

earboxer commented:
@lpugin I changed it to target develop-transpose, and also added `#include <cctype>`, which *should* allow [isdigit](https://en.cppreference.com/w/cpp/string/byte/isdigit) to work.

Do we need to change [`appveyor.yml`](https://github.com/rism-ch/verovio/blob/develop/appveyor.yml) so that it runs on develop-transpose?

craigsapp commented:
Should I CamelCase the functions in the `Transpose` class?  Perhaps not necessary since the class is not directly related to MEI functionality?

lpugin commented:
> Should I CamelCase the functions in the Transpose class? Perhaps not necessary since the class is not directly related to MEI functionality?

Yes please. (This is a Verovio coding guideline)

craigsapp commented on include/vrv/note.h:
```diff
@@ -182,6 +184,13 @@ class Note : public LayerElement,
+     */
+    int GetChromaticAlteration();
+
+    TPitch GetTPitch();
```
Yes, that is good.

craigsapp replied:
Done.

craigsapp commented on include/vrv/transpose.h:
```diff
@@ -0,0 +1,34 @@
+class TPitch;
+
+class TPitch {
+public:
```
In this case the member variables are public, so it seems less appropriate for such variables to be prefixed by `m_` (TPitch acts more like a struct rather than a class).   I could make the variables private and access with member functions (that would make `m_` prefix more appropriate).

lpugin replied:
I am fine having them public but I would still prefer to have them prefixed. (I know there are a few cases where this is not the case, such as with Point)

craigsapp replied:
Done.

craigsapp commented on include/vrv/note.h:
```diff
@@ -21,13 +21,15 @@
 #include "durationinterface.h"
 #include "layerelement.h"
 #include "pitchinterface.h"
+#include "transpose.h"
```
Done.  Should I also rename the class `Transposition`?

craigsapp replied:
Best to rename in any case since `Transpose::Transpose` is a constructor, so cannot CamelCase `Transpose::transpose` otherwise.

craigsapp replied:
Done.

lpugin replied:
I would have suggested `Transposer` (similar to various *Aligner classes for layout)

craigsapp replied:
Done.
